pkg/controller/servicebindingrequest/binder.go

```diff
@@ -237,26 +239,39 @@ func (b *binder) updateSpecContainers(
 	if containers, err = b.updateContainers(containers); err != nil {
 		return nil, err
 	}
-	if err = unstructured.SetNestedSlice(obj.Object, containers, containersPath...); err != nil {
+	if err = unstructured.SetNestedSlice(obj.Object, containers, b.getContainersPath()...); err != nil {
 		return nil, err
 	}
 	return obj, nil
 }
 
+func (b *binder) getContainersPath() []string {
+	return strings.Split(b.sbr.Spec.ApplicationSelector.BindingPath.PodSpecPath.Containers, ".")
```

Comment:
is there a chance that this is called on not initialized value?

Reply:
> is there a chance that this is called on not initialized value?

I don't understand the change required here.

Reply:
As pointed out in [a comment](https://github.com/redhat-developer/service-binding-operator/pull/513/files#r459310502) related to `SetDefaults`, `b.sbr.Spec.ApplicationSelector.BindingPath` can be `nil` in this statement.

Reply:
Fixed. 8471a6520097095587857d584c3df8585112f11c

Reply:
Moving under the api makes it a public API. I think that's not really necessary.

Reply:
Why not, can you explain why do you think that adding such getter on public API would not be beneficial?  What do you want to hide with this private method? Do we want to process `Containers` values in several ways?